.github/workflows/erlang-parallel-build.yml

```
@@ -0,0 +1,223 @@
+          generator-version: ${{ inputs.erlang-generator-version }}
+
+      - name: Cache _build
+        uses: actions/cache@v2
+        with:
+          path: _build/*/lib
+          key: ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-${{ hashFiles('rebar.lock') }}
+          restore-keys: |
+            ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-
+
+      - name: Compile
+        run: |
+          rebar3 compile
+          rebar3 as test compile
+
+  check:
+    name: Check
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup BEAM
+        uses: erlef/setup-beam@v1.10
+        with:
+          otp-version: ${{ inputs.otp-version }}
+          rebar3-version: ${{ inputs.rebar-version }}
+
+      - name: Cache _build
+        uses: actions/cache@v2
+        with:
+          path: _build/*/lib
+          key: ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-${{ hashFiles('rebar.lock') }}
+          restore-keys: |
+            ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-
+
+      - name: Check formatting
+        run: rebar3 fmt -c
+
+      - name: Run linting
+        run: rebar3 lint
+
+      - name: Run xref
+        run: rebar3 xref
+
+  dialyze:
+    name: Dialyze
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup BEAM
+        uses: erlef/setup-beam@v1.10
+        with:
+          otp-version: ${{ inputs.otp-version }}
+          rebar3-version: ${{ inputs.rebar-version }}
+
+      - name: Cache _build
+        uses: actions/cache@v2
+        with:
+          path: _build/*/lib
+          key: ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-${{ hashFiles('rebar.lock') }}
+          restore-keys: |
+            ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-build-
+
+      - name: Cache PLTs
+        uses: actions/cache@v2
+        with:
+          path: _build/test/rebar3_*_plt
+          key: ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-plt-${{ hashFiles('rebar.lock') }}
+          restore-keys: |
+            ${{ inputs.cache-version }}-${{ runner.os }}-otp-${{ inputs.otp-version }}-plt-      
+
+      - name: Run dialyzer
+        run: rebar3 as test dialyzer
+
+  test:
+    name: Test
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup BEAM
+        uses: erlef/setup-beam@v1.10
+        with:
+          otp-version: ${{ inputs.otp-version }}
+          rebar3-version: ${{ inputs.rebar-version }}
+
+      - name: Cache _build
+        uses: actions/cache@v2
+        with:
+          path: _build/*/lib
```

Быстрее будет качать кокретный профиль.
```suggestion
          path: _build/test/lib
```

---

Судя по сорсцам actions/cache кеш все равно хранится в одном архиве, так что врядли тут будет разница. Я потестирую

---

Эксперементальным путем установил, что test профиль все же зависит от default (хотя я тоже раньше почему то считал, что они самодостататочны), и отсутсвие default [вызывает рекомпиляцию](https://github.com/valitydev/token-keeper/runs/4963834857?check_suite_focus=true).